bufbuild/buf.bzl

```diff
@@ -0,0 +1,43 @@
+load("@rules_proto//proto:defs.bzl", "proto_common")
+
+def _buf_gen_impl(ctx):
+    buf_toolchain = ctx.toolchains["//:buf_toolchain_type"]
+
+    args = ctx.actions.args()
+    args.add("generate")
+    args.add(ctx.file.template, format = "--template=%s")
+
+    inputs = [ctx.file.template]
+    outputs = []
+    for dep in ctx.attr.deps:
+        outputs.extend(
+            proto_common.declare_generated_files(
```

Comment:
Should it be Go-specific? Would it be possible you guide us how to generate pb files for other languages?

Reply:
Yeah this example is Go-specific. We can definitely show you other languages, but we don't have time to do that for free. Consider https://www.aspect.dev/bazel-open-source-support or fund with the Sponsor link on this repo.